MetaMask/metamask-mobile
# fix: Warn users when connecting to a website on the eth-phishing-detect list on mobile

## **Description**
Currently users don't see any warning when connecting to a site on the eth-phishing-detect list using wallet connect. This PR checks if the url is flagged in the phishing controller and shows a warning banner when connecting to such site.

## **Manual testing steps**

1. Setup [WC](https://github.com/WalletConnect/create-wc-dapp) locally using `npx create-wc-dapp@latest web3modal-quickstart -id <PRPOJECT_ID> -y`
2. Run your local server
3. In line 36 of `src/pages/_app.tsx`, change the default url in to a flagged url from the phishing controller
4. Connect to the server on your device and connect to wallet.
5. If url in the phishing list, it should show the warning banner on the connect modal else no warning banner.


Bitrise: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/44a67e4f-4e42-4959-8cba-a9f25c45401b


## **Screenshots/Recordings**

### **Before**


https://github.com/MetaMask/metamask-mobile/assets/29962968/fdbbbbb6-e783-43f8-b0ed-70b3a8c2b0f1


### **After**


https://github.com/MetaMask/metamask-mobile/assets/29962968/4134793e-1d9e-42cf-b9f0-ef4979f11e50






## **Related issues**

_Fixes https://github.com/MetaMask/MetaMask-planning/issues/1327

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
